spring-cloud-data-rest-client/src/main/java/org/springframework/cloud/data/rest/client/ModuleOperations.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.rest.client;
+
+import org.springframework.cloud.data.core.ModuleType;
+import org.springframework.cloud.data.rest.resource.DetailedModuleRegistrationResource;
+import org.springframework.cloud.data.rest.resource.ModuleRegistrationResource;
+import org.springframework.hateoas.PagedResources;
+
+/**
+ * Interface defining operations available for module registrations.
+ *
+ * @author Glenn Renfro
+ * @author Gunnar Hillert
+ * @author Eric Bottard
+ * @author Patrick Peralta
+ */
+public interface ModuleOperations {
+
+	/**
+	 * Return a list of all module registrations.
+	 *
+	 * @return list of all module registrations
+	 */
+	PagedResources<ModuleRegistrationResource> list();
+
+	/**
+	 * Return a list of all module registrations for the given {@link ModuleType}.
+	 *
+	 * @param type module type for which to return a list of registrations
+	 * @return list of all module registrations for the given module type
+	 */
+	PagedResources<ModuleRegistrationResource> list(ModuleType type);
+
+	/**
+	 * Retrieve information about a module registration.
+	 *
+	 * @param name name of module
+	 * @param type module type
+	 *
+	 * @return detailed information about a module registration
+	 */
+	DetailedModuleRegistrationResource info(String name, ModuleType type);
+
+	/**
+	 * Register a module name and type with its Maven coordinates.
+	 *
+	 * @param name  module name
+	 * @param type  module type
+	 * @param coordinates  Maven coordinates for the module artifact
+	 * @param force if {@code true}, overwrites a pre-existing registration
+	 */
+	ModuleRegistrationResource registerModule(String name, ModuleType type,
```

Comment:
For consistency (with other methods in here as well as the methods of the ModuleController), we should drop "Module" from the `register`/`unregister` method names. Also relevant, see this change we agreed on earlier today for TaskOperations: https://github.com/cppwfs/spring-cloud-data/commit/ab4f6b5ad15b3f4123d8911ba0740bb118f9720c#diff-9a647bdc0cb461b538bff06ed9c338efL39